sqlx-mysql/src/types/geometry.rs

```diff
@@ -0,0 +1,41 @@
+use geo_types::Geometry;
+use geozero::{wkb::{FromWkb, WkbDialect, WkbWriter}, CoordDimensions, GeozeroGeometry, ToWkb};
+
+use crate::decode::Decode;
+use crate::encode::{Encode, IsNull};
+use crate::error::BoxDynError;
+use crate::protocol::text::ColumnType;
+use crate::types::Type;
+use crate::{MySql, MySqlTypeInfo, MySqlValueRef};
+use crate::io::MySqlBufMutExt;
+
+// Only Geometry<f64> is implemented with geozero::GeozeroGeometry.
+impl Type<MySql> for Geometry<f64> {
+    fn type_info() -> MySqlTypeInfo {
+        MySqlTypeInfo::binary(ColumnType::Blob)
+    }
+
+    fn compatible(ty: &MySqlTypeInfo) -> bool {
+        ty.r#type == ColumnType::Geometry || <&[u8] as Type<MySql>>::compatible(ty)
+    }
+}
+
+const ENCODE_ERR: &str = "failed to encode value as Geometry to WKB";
```

Comment:
maybe consider something like "why this wouldn't fail" if this can't fail, or avoid panicking

Reply:
I'm trying to do something similar to https://github.com/launchbadge/sqlx/blob/main/sqlx-core/src/types/json.rs#L102.

Maybe I need to add more messages like "most likely cause" or "how to fix it".